app/internal/notification.py

```diff
@@ -0,0 +1,174 @@
+from operator import attrgetter
+from typing import List, Union
+
+from fastapi import HTTPException
+from sqlalchemy.exc import SQLAlchemyError
+from sqlalchemy.orm import Session
+from starlette.status import HTTP_401_UNAUTHORIZED, HTTP_406_NOT_ACCEPTABLE
+
+from app.database.models import (
+    Invitation,
+    Message,
+    InvitationStatusEnum,
+    MessageStatusEnum,
+)
+from app.internal.utils import create_model
+
+
+WRONG_NOTIFICATION_ID = (
+    "The notification id you have entered is wrong\n."
+    "If you did not enter the notification id manually, report this exception."
+)
+
+NOTIFICATION_TYPE = Union[Invitation, Message]
+
+UNREAD_STATUS = [
+    InvitationStatusEnum.UNREAD,
+    MessageStatusEnum.UNREAD,
+]
+
+ARCHIVED = [
+    InvitationStatusEnum.DECLINED,
+    MessageStatusEnum.READ,
+]
+
+
+async def get_message_by_id(
+    message_id: int,
+    session: Session,
+) -> Union[Message, None]:
+    """Returns an invitation by an id.
+    if id does not exist, returns None."""
+    return session.query(Message).filter_by(id=message_id).first()
+
+
+def _is_unread(notification: NOTIFICATION_TYPE) -> bool:
+    """Returns True if notification is unread, False otherwise."""
+    return notification.status in UNREAD_STATUS
+
+
+def _is_archived(notification: NOTIFICATION_TYPE) -> bool:
+    """Returns True if notification should be
+    in archived page, False otherwise.
+    """
+    return notification.status in ARCHIVED
+
+
+def is_owner(user, notification: NOTIFICATION_TYPE) -> bool:
+    """Checks if user is owner of the notification.
+
+    Args:
+        notification: a NOTIFICATION_TYPE object.
+        user: user schema object.
+
+    Returns:
+        True or raises HTTPException.
+    """
+    if notification.recipient_id == user.user_id:
+        return True
+
+    msg = "The notification you are trying to access is not yours."
+    raise HTTPException(
+        status_code=HTTP_401_UNAUTHORIZED,
+        detail=msg,
+    )
+
+
+def raise_wrong_id_error() -> None:
+    """Raises HTTPException.
+
+    Returns:
+        None
+    """
+    raise HTTPException(
+        status_code=HTTP_406_NOT_ACCEPTABLE,
+        detail=WRONG_NOTIFICATION_ID,
+    )
+
+
+def filter_notifications(
+    session: Session,
+    user_id: int,
+    func: Union[_is_unread, _is_archived],
```

Comment:
`_is_unread` and `is_archived` are not a type. Use `Callable`. See PEP 484 or mypy documentations for more details.
Prefer validating your code using mypy.